### ISSUE TITLE
Changing licence of tooling to AGPLv3

### ISSUE THREAD

**Issue by iHiD** [exercism/exercism#5334]
## TL;DR;

We are changing the licence of our Tooling (test runners, representers, analyzers, etc) from MIT to AGPL. All new Tooling will be released under AGPL. Track repositories remain MIT.

## Background

Exercism's website and supporting software has always been licensed under APGL. Our tracks have always been released under MIT. When we started building Tooling for the first time last year, we didn't put enough thought into the licence we should use for it, and chose MIT for consistency with our tracks. We've now realised that's a bad idea, and so are changing the licence of Tooling to be inline with the website, not the track content.

## Rational

Exercism is a free, open-source platform, and our community build Exercism because of those decisions. The MIT licence allows someone to take a copy of Exercism's source code, modify it and start charging for it, without giving back those improvements to Exercism. It would, for example, allow a large organisation to fork Exercism, invest VC funds into that fork to develop it, and commercialise it, without giving back to the open source community or to Exercism [1].

The [AGPL v3 Licence](https://www.gnu.org/licenses/agpl-3.0.en.html) protects against this as it enforces that any modifications to the code of our Tooling must themselves be open-sourced. The preamble of this licence states that it is designed "to guarantee your freedom to share and change all versions of a program - to make sure it remains free software for all its users." As the purpose of Exercism's Tooling is for use within Exercism, and not within some commercial closed-source project, we have decided to add the protections of AGPL on an ongoing basis.

## Methodology

The easiest way for us to do this is to ask the existing contributors to relicense their code from MIT to AGPL. I will be following with PRs in each repository, asking the contributors to assert their agreement to this, and where they all do, we will simply change the licence. In situations where we cannot solicit agreement from a significant amount of contributors, we will change the licence to AGPL, but also add a notice explaining this change along with the original MIT licence (known as sublicensing) [2]. 

## Notes

[1] For some more in depth discussion on this, I recommend reading "How to Capture an Open Source Project" from Chapter 2 of Pieter Hintjens' "Social Architecture", which discusses how a company can use an MIT licence to effectively close off an open source project. A late draft of the book is readable online [here](https://github.com/hintjens/socialarchitecture/blob/5df33c4d533e280a9c52707dc1b23c4e7e954f34/ch02.txt). It also references [this post](https://arstechnica.com/gadgets/2018/07/googles-iron-grip-on-android-controlling-open-source-by-any-means-necessary/) subtitled "Android is open—except for all the good parts"

[2] Mozilla's lawyers determined that an author coverage of 95% to be sufficient when they made the same transition.

**Comment by ErikSchierboom** [comment]
Hello 👋

With the launch of Exercism v3, we are **closing all issues in this repository** to help give us a clean slate to detect new problems. If this issue is still relevant to Exercism v3 (e.g. it's a feature that we haven't implemented in v3, or a bug that still exists), please reopen it and we will review it and post an update on it as soon as we get chance.

Thanks for helping make Exercism better, and we hope you enjoy v3 🙂
